app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -408,6 +417,46 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
         }
     }
 
+    private fun applyMediaSegments(jellyfinMediaSource: JellyfinMediaSource) {
+        viewModelScope.launch {
+            if (jellyfinMediaSource.item != null) {
+                val mediaSegments = runCatching {
+                    mediaSegmentRepository.getSegmentsForItem(jellyfinMediaSource.item)
+                }.getOrNull().orEmpty()
```

Comment:
This shouldn't throw, since the function already uses runCatching. Can thus be simplified.